tutorials/dataframe/df106_HiggsToFourLeptons_withVariations.py

```diff
@@ -118,53 +130,115 @@
+        return RVec<double>{(1 + v) * w, (1 - v) * w};
+    }
+};
+
+VaryHelper myHelper;
+""")
+
+# Use the Vary method to add the systematic variations to the total MC scale factor ("weight") of the analysis. 
+df_variations_mc = df.Filter("weight != 1.0")\
+                     .Vary("weight", "myHelper(weight, goodlep_pt, goodlep_type)", ["up", "down"])\
+                     .Histo1D(ROOT.RDF.TH1DModel("Invariant Mass", "m4l", 24, 80, 170), "m4l", "weight")                  
+histos_mc = ROOT.RDF.Experimental.VariationsFor(df_variations_mc)                                 
+
+# Make the plot of the data, individual MC contributions and the total MC scale factor systematic variations. 
 
 # Set styles
 ROOT.gROOT.SetStyle("ATLAS")
 
+# Function to add ATLAS label
+def ATLAS_label():
+    text = ROOT.TLatex()
+    text.SetNDC()
+    text.SetTextFont(72)
+    text.SetTextSize(0.045)
+    text.DrawLatex(0.21, 0.86, "ATLAS")
+    text.SetTextFont(42)
+    text.DrawLatex(0.21 + 0.15, 0.86, "Open Data")
+    text.SetTextSize(0.04)
+    text.DrawLatex(0.21, 0.80, "#sqrt{s} = 13 TeV, 10 fb^{-1}")
+
 # Create canvas with pad
-c = ROOT.TCanvas("c", "", 600, 600)
+c1 = ROOT.TCanvas("c", "", 600, 600)
 pad = ROOT.TPad("upper_pad", "", 0, 0, 1, 1)
 pad.SetTickx(False)
 pad.SetTicky(False)
 pad.Draw()
 pad.cd()
 
 # Draw stack with MC contributions
-
 stack = ROOT.THStack()
 
-for h, color in zip([h_other, h_zz, h_higgs], [(155, 152, 204), (100, 192, 232), (191, 34, 41)]):
+# Retrieve values of the data and MC histograms in order to plot them. 
+# Note: GetValue() action operation is performed after all lazy actions of the RDF were defined first. 
+h_data = histos[0].GetValue()
+h_higgs = histos[1].GetValue()
+h_zz = histos[2].GetValue()
+h_other = histos[3].GetValue()
+
+for h, color in zip([h_other, h_zz, h_higgs], [ROOT.kViolet-9, ROOT.kAzure-9, ROOT.kRed+2]):
     h.SetLineWidth(1)
     h.SetLineColor(1)
-    h.SetFillColor(ROOT.TColor.GetColor(*color))
+    h.SetFillColor(color)
     stack.Add(h)
 
 stack.Draw("HIST")
 stack.GetXaxis().SetLabelSize(0.04)
 stack.GetXaxis().SetTitleSize(0.045)
 stack.GetXaxis().SetTitleOffset(1.3)
 stack.GetXaxis().SetTitle("m_{4l}^{H#rightarrow ZZ} [GeV]")
-stack.GetYaxis().SetTitle("Events")
 stack.GetYaxis().SetLabelSize(0.04)
 stack.GetYaxis().SetTitleSize(0.045)
-stack.SetMaximum(33)
+stack.GetYaxis().SetTitle("Events")
+stack.SetMaximum(35)
 stack.GetYaxis().ChangeLabel(1, -1, 0)
 
+# Draw MC scale factor and variations 
+(histos_mc)["nominal"].SetLineColor(ROOT.kOrange+10)
+(histos_mc)["nominal"].SetStats(0)
+(histos_mc)["nominal"].Draw("HIST SAME")
+(histos_mc)["weight:up"].SetStats(0)
+(histos_mc)["weight:up"].SetLineColor(ROOT.kGreen+2)
+(histos_mc)["weight:up"].Draw("HIST SAME")
+(histos_mc)["weight:down"].SetLineColor(ROOT.kBlue+2)
+(histos_mc)["weight:down"].SetStats(0)
+(histos_mc)["weight:down"].DrawCopy("HIST SAME")
```

Comment:
```suggestion
histos_mc["nominal"].SetLineColor(ROOT.kOrange+10)
histos_mc["nominal"].SetStats(0)
histos_mc["nominal"].Draw("HIST SAME")
histos_mc["weight:up"].SetStats(0)
histos_mc["weight:up"].SetLineColor(ROOT.kGreen+2)
histos_mc["weight:up"].Draw("HIST SAME")
histos_mc["weight:down"].SetLineColor(ROOT.kBlue+2)
histos_mc["weight:down"].SetStats(0)
histos_mc["weight:down"].DrawCopy("HIST SAME")
```

We really need the parenthesis here?

Reply:
you're correct, we don't